x-pack/plugins/security_solution/public/flyout/right/hooks/use_fetch_unique_hosts_with_field_value_pair.ts

```diff
@@ -0,0 +1,138 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { buildEsQuery } from '@kbn/es-query';
+import type { IEsSearchRequest } from '@kbn/data-plugin/public';
+import { useQuery } from '@tanstack/react-query';
+import { createFetchAggregatedData } from '../utils/fetch_aggregated_data';
+import { useKibana } from '../../../common/lib/kibana';
+import { inputsSelectors } from '../../../common/store';
+import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { useGlobalTime } from '../../../common/containers/use_global_time';
+
+const AGG_KEY = 'hostsWithSameFieldValuePair';
+const QUERY_KEY = 'useFetchUniqueHostsWithFieldPair';
+
+interface RawAggregatedDataResponse {
+  aggregations: {
+    [AGG_KEY]: {
+      buckets: unknown[];
+    };
+  };
+}
+
+export interface UseFetchUniqueHostWithFieldPairParams {
+  /**
+   * Highlighted field
+   */
+  field: string;
+  /**
+   * Highlighted field value
+   */
+  values: string[];
+  /**
+   *
+   */
+  isActiveTimelines: boolean;
+}
+
+export interface UseFetchUniqueHostWithFieldPairResult {
+  /**
+   * Returns true if data is being loaded
+   */
+  loading: boolean;
+  /**
+   * Returns true if fetching data has errored out
+   */
+  error: boolean;
+  /**
+   * Number of unique hosts found for the field/value pair
+   */
+  count: number;
+}
+
+/**
+ * Hook to retrieve all the unique hosts in the environment that have the field/value pair, using ReactQuery.
+ * The query uses an aggregation by unique hosts.
+ */
+export const useFetchUniqueHostsWithFieldPair = ({
+  field,
+  values,
+  isActiveTimelines,
+}: UseFetchUniqueHostWithFieldPairParams): UseFetchUniqueHostWithFieldPairResult => {
+  const {
+    services: {
+      data: { search: searchService },
+    },
+  } = useKibana();
+
+  const timelineTime = useDeepEqualSelector((state) =>
+    inputsSelectors.timelineTimeRangeSelector(state)
+  );
+  const globalTime = useGlobalTime();
+  const { to, from } = isActiveTimelines ? timelineTime : globalTime;
+
+  const query = buildEsQuery(
+    undefined,
+    [],
+    [
+      {
+        query: {
+          bool: {
+            filter: [
+              {
+                match: {
+                  [field]: values[0],
+                },
+              },
+              {
+                range: {
+                  '@timestamp': {
+                    gte: from,
+                    lte: to,
+                  },
+                },
+              },
+            ],
+          },
+        },
+        meta: {},
+      },
+    ]
+  );
+  const req: IEsSearchRequest = {
+    params: {
+      body: {
+        query,
+        aggs: {
+          [AGG_KEY]: {
+            terms: {
+              field: 'host.name',
+              size: 1000,
+            },
+          },
+        },
+        size: 0,
+      },
+    },
+  };
```

Comment:
could this be moved into a function of its own?

Reply:
Done! Didn't think it was necessary, but now that I've done it I like how much cleaner the method is, thanks!